### PR TITLE
Collector refactoring, part 2

### PR DESCRIPTION
**Pull request by Kobzol**
Now the `collector` crate has a reasonable module layout (at least to my taste), the largest files were split and all modules are under the library root, the `collector` binary only uses them. There is of course still a lot of legacy cruft in some of the interfaces, but I think that for now the refactoring effort is mostly completed from my side. I'll send more patches if needed for runtime benchmarks or benchmark error reporting.

### CODE REVIEW

File: collector/Cargo.toml [hunk 1]
```diff
@@ -6,7 +6,7 @@ edition = "2018"
 description = "Collects Rust performance data"
 
 [dependencies]
-clap = { version = "3.0.6", features = ["derive"] }
```
Comment (rylev) [inline collector/Cargo.toml:9]: Would it make sense for us to do `version = 3.2`?

File: collector/src/benchmark/scenario.rs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    pub fn all_non_incr() -> Vec<Scenario> {
+        vec![Scenario::Full]
+    }
+
+    pub fn is_incr(self) -> bool {
+        match self {
+            Scenario::Full => false,
+            Scenario::IncrFull | Scenario::IncrUnchanged | Scenario::IncrPatched => true,
+        }
```
Comment (rylev) [inline collector/src/benchmark/scenario.rs:28]:
```rust
matches!(self, Scenario::IncrFull | Scenario::IncrUnchanged | Scenario::IncrPatched)
```

**Review by rylev — APPROVED** [review]
This seems like a good refactoring. I tend to prefer medium size files (as opposed to the long files we had before or another approach where we have one file per type), and this seems to strike a good balance.

I'm interested what others have to say.

**Comment by nnethercote** [comment]
I don't mind longer files, e.g. anything less than 2,000 lines and I don't bat an eyelid. So a lot of this doesn't feel that necessary to me. But I also won't stand in the way, particular given that Jakub (presumably) thinks this will help with the addition of runtime benchmarks.

**Comment by Kobzol** [comment]
It wasn't necessarily only the length that was annoying to me. The main issue for me was that some modules were included from the binary instead of the library root (this was annoying especially for the Profile and Scenario enums, which are used all over the place and prohibited using collector as a library properly).

Also, some files were a bit too overarching. A 2k line file is fine if it cares about a single area, but less so if it spans X (only semi-)related things that could be its own modules.

File: collector/Cargo.toml (hunk 1 above)
Comment (Kobzol) [reply collector/Cargo.toml:9]: Yes, I changed it.

**Pull request merged**